features/networking/pod.feature

```diff
@@ -190,3 +190,126 @@ Feature: Pod related networking scenarios
     Then the step should succeed
     And the output should not contain "ingress_policing_rate: 1953"
 
+  # @auther anusaxen@redhat.com
+  # @case_id OCP-23890
+  @admin
+  Scenario: A pod with or without hostnetwork cannot access the MCS port 22623 or 22624 on the master
+    Given I use the first master host
+    #Step to obtain master IP
+    And I run commands on the host:
+      | hostname -i |
+    Then the step should succeed
+    And evaluation of `@result[:response].strip` is stored in the :master_ip clipboard
+    
+    Given I select a random node's host
+    Given I have a project
+    #pod-for-ping will be a non-hostnetwork pod
+    And SCC "privileged" is added to the "system:serviceaccounts:<%= project.name %>" group
+    And I have a pod-for-ping in the project
+    And the pod named "hello-pod" becomes ready
+
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22623/config/master | -k |
+    Then the output should contain "Connection refused"
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22624/config/master | -k |
+    Then the output should contain "Connection refused"
+    
+    #hostnetwork-pod will be a hostnetwork pod
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/anuragthehatter/v3-testfiles/master/networking/hostnetwork-pod.json |
+      | n | <%= project.name %>                                                                                   |
+    Then the pod named "hostnetwork-pod" becomes ready
+    #Pods should not access the MCS port 22623 or 22624 on the master
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22623/config/master | -k |
+    Then the output should contain "Connection refused"
+    When I execute on the pod:
+      | curl | -I | https://<%= cb.master_ip %>:22624/config/master | -k |
+    Then the output should contain "Connection refused"
+
+  # @auther anusaxen@redhat.com
+  # @case_id OCP-23891
+  @admin
+  Scenario: A pod cannot access the MCS port 22623 or 22624 via the SDN/tun0 address of the master
+    Given I use the first master host		
+    And I run commands on the host:
+      | ifconfig tun0 \| grep -w inet \| awk '{print $2}' |
```

Comment:
can we instead create a loop getting all local IPs on master and checking them? This catching every interface, that may change with future releases sounds like fragile.

Reply:
Thanks for idea Alex! What i see is tun0 as a special interface and i don't think so its going to be disappear in future due to requirement by OVS and OVN too which is upcoming tech. Lot of logic built around it in traffic circulation among cluster. I guess even if i try to create loop for getting all local IPs, i eventually have to search for tun0 and its also a specific requirement by user story as we see in scenario. wdyt

Reply:
ok,  I trust your assessment.